src/main/java/org/apache/sysml/parser/pydml/PydmlSyntacticValidator.java

```diff
@@ -959,9 +959,11 @@ else if(namespace.equals(DMLProgram.DEFAULT_NAMESPACE) && functionName.equals("r
 			namespace = DMLProgram.DEFAULT_NAMESPACE;
 		}
 		else if(namespace.equals(DMLProgram.DEFAULT_NAMESPACE) && functionName.equals("range")) {
-			if(paramExpression.size() != 3) {
```

Comment:
@niketanpansare - can you please take a look at this change?
The semantics of the `range` function in pydml are being changed here. The change makes sense to me, but seeing as how you spent the most amount of time working on PyDML, do you think this is ok?